pkg/controller/controller.go

```diff
@@ -183,9 +186,29 @@ func New(ctx context.Context) {
 	}
 }
 
-func createController(kubeClient kubernetes.Interface, informer cache.SharedIndexInformer, resource string) *KubeResourceWatcher {
+func getRateLimitTime() int {
+	rateString := os.Getenv("RATELIMIT_INTERVAL")
+	if rateString != "" {
+		rateInt, err := strconv.Atoi(rateString)
+		if err != nil {
+			log.Warn("RATELIMIT_INTERVAL not set to a valid integer value", err)
+			return 500
+		}
+		return rateInt
+	}
+	return 500
+}
+
+func createController(kubeClient kubernetes.Interface, informer cache.SharedIndexInformer, resource string, rateLimit int) *KubeResourceWatcher {
 	log.Debugf("Creating controller for resource type %s", resource)
-	wq := workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter())
+	rateLimiter := workqueue.NewMaxOfRateLimiter(
+		// Default is 5 Millisecond base and a max of 1000 seconds.
+		// Lowered to 100 second max
+		// Raised the base time to 500 millieconds * number of tries squared.
+		workqueue.NewItemExponentialFailureRateLimiter(time.Duration(rateLimit)*time.Millisecond, 100*time.Second),
```

Comment:
Is 100 seconds max enough? In the issue, there is an error with 5 puts within 600 seconds. Would we drop the 6th request in this case?

Reply:
Wow! That's an aggressive rate limit on Datadog's part. Maybe we should raise how the exponential backoff starts at. We shouldn't actually hit that 100 seconds anyways. With the default of a half second it would go like this.

Initial try
0.5 seconds later retry 1
2 seconds later retry 2
4.5 seconds later retry 3
8 seconds later retry 4
12.5 seconds later retry 5.
If that fails then we have a 5 retry max.

In an example to hit the 100 it would look like this. If the base time was set to 20 seconds

Initial try
20 seconds later retry 1
80 seconds later retry 2
100 seconds later retry 3
100 seconds later retry 4
100 seconds later retry 5.

It's still probably worth raising that up though if Datadog's limit was really 5 within 600.

Reply:
I also think the issue may refer to a DD account with super low rate limiting for some reason (maybe a new account for testing?). I've never seen a rate limit that aggressive in my testing, but we may have had it increased at some point for our internal account(s).

Reply:
I experienced the same(`Rate limit of 5 requests in 600 seconds reached. `) on a non-testing account. The rate limit was too low. I had to ask DD to increase the rate limit.